### ISSUE TITLE
Svg filter not rendered correctly

### ISSUE THREAD

**Issue by royvs33** [luberda-molinet/FFImageLoading#1205]
### Description

The SVG as shown below does not seem te render correctly, the problem is in the filters, but cannot find out why. I have tested SVGOMG but it does not seem to fixed the issue

### Steps to Reproduce

Make an SvgCachedImage element with the SVG as shown below, then it renders black

### Expected Behavior

![image](https://user-images.githubusercontent.com/7634471/53637768-57245980-3c24-11e9-98d4-bd9f3af05655.png)

### Actual Behavior

![image](https://user-images.githubusercontent.com/7634471/53637825-7de29000-3c24-11e9-9932-2fa4492e489a.png)

### Basic Information

- Version with issue: 2.4.3.840 (also tried with 2.4.4.859)
- Platform: Android

### Reproduction Link / Code

```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="240" height="51">
  <defs>
    <path id="a" d="M25.5 0h189a25.5 25.5 0 0 1 0 51h-189a25.5 25.5 0 0 1 0-51z"/>
    <filter id="b" width="103.3%" height="115.7%" x="-1.7%" y="-7.8%" filterUnits="objectBoundingBox">
      <feGaussianBlur in="SourceAlpha" result="shadowBlurInner1" stdDeviation="3"/>
      <feOffset dy="2" in="shadowBlurInner1" result="shadowOffsetInner1"/>
      <feComposite in="shadowOffsetInner1" in2="SourceAlpha" k2="-1" k3="1" operator="arithmetic" result="shadowInnerInner1"/>
      <feColorMatrix in="shadowInnerInner1" values="0 0 0 0 0.142777423 0 0 0 0 0.0763265436 0 0 0 0 0.0763265436 0 0 0 0.527004076 0"/>
    </filter>
  </defs>
  <g fill="none" fill-rule="evenodd">
    <use fill="#E16070" xlink:href="#a"/>
    <use fill="#000" filter="url(#b)" xlink:href="#a"/>
  </g>
</svg>
```

**Comment by daniel-luberda** [comment]
Filters are not supported yet. We'll need to implement it.

**Comment by stfnilsson** [comment]
I use my own svg-skiaharp code to render svg where I need to fill, but I want to use only your code.
Can't you add a parameter for the fill/paint color to SvgCachedImage, I want to use it via XAML..

**Comment by daniel-luberda** [comment]
@stfnilsson Could you describe in details what you want to achieve? Do you want to change colors? If yes, then maybe `ReplaceStringMap` property would fit you (you can replace any SVG part with it to your custom text)

**Comment by stfnilsson** [comment]
I want to fill my svg icons with a color, I use SvgCachedImage (XAML). The colors are defined in a style.xaml. 

TBut I will just created a control with a Color property and fix the ReplaceStringMap inside the control, maybe even better with a converter. Maybe I put the ReplaceStringMap for the different colors inside style.xaml.

Is that something for your SvgCachedImage? I understand that you want it clean but what would make it much easier to use SvgCachedImage (not saying ReplaceStringMap is hard)

**Comment by daniel-luberda** [comment]
For now, you can simply inherit from `SvgCachedImage` and add something like this to your custom control: 

```C#
public static BindableProperty ColorProperty = BindableProperty.Create(nameof(Color), typeof(Color), typeof(SvgCachedImage));
public Color Color
{
    get { return (Color)GetValue(ColorProperty); }
    set { SetValue(ColorProperty, value); }
}

protected override void OnPropertyChanged([CallerMemberName] string propertyName = null)
{
    base.OnPropertyChanged(propertyName);

    if (propertyName == ColorProperty.PropertyName)
    {
        ReplaceStringMap = new Dictionary<string, string>() { { "#000000", Color.ToString() } }
    }
}
```

BTW: you can also use replace string map per image source (it's in `SvgImageSource` constructors)

It's possible to implement replacing all colors in SVG, but it's not done yet. For now, I'm focused on this: https://github.com/luberda-molinet/FFImageLoading/issues/1280 as it will give us greatly improved SVG support.

**Comment by stfnilsson** [comment]
Yes, liked I planned. I started with a converter but can’t remember why it didn’t work.

My suggestion was just to add it to SvgCachedImage so it will be even more easier to use the control.

Thanks for a great job with FFImageLoading 😊


Skickades från E-post<https://go.microsoft.com/fwlink/?LinkId=550986> för Windows 10

________________________________
Från: Daniel Luberda <notifications@github.com>
Skickat: Thursday, May 30, 2019 1:35:44 PM
Till: luberda-molinet/FFImageLoading
Kopia: Stefan Nilsson; Mention
Ämne: Re: [luberda-molinet/FFImageLoading] SVG filter fill support (#1205)


For now, you can simply inherit from SvgCachedImage and add something like this to your custom control:

public static BindableProperty ColorProperty = BindableProperty.Create(nameof(Color), typeof(Color), typeof(SvgCachedImage));
public Color Color
{
    get { return (Color)GetValue(ColorProperty); }
    set { SetValue(ColorProperty, value); }
}

protected override void OnPropertyChanged([CallerMemberName] string propertyName = null)
{
    base.OnPropertyChanged(propertyName);

    if (propertyName == ColorProperty.PropertyName)
    {
        ReplaceStringMap = new Dictionary<string, string>() { { "#000000", Color.ToString() } }
    }
}

BTW: you can also use replace string map per image source (it's in SvgImageSource constructors)

It's possible to implement replacing all colors in SVG, but it's not done yet. For now, I'm focused on this: #1280<https://github.com/luberda-molinet/FFImageLoading/issues/1280> as it will give us greatly improved SVG support.

**Comment by requenaxii** [comment]
Same issue here, also with `LinearGradient`, any uptade or workaround?

This is my SVG defs: 

```xml
    <defs>
        <linearGradient id="c" x1="50%" x2="50%" y1="0%" y2="100%">
            <stop offset="0%" stop-color="#00ACEC"/>
            <stop offset="32.769%" stop-color="#00ABEB"/>
            <stop offset="100%" stop-color="#4370B6"/>
        </linearGradient>
        <rect id="a" width="90" height="52" rx="4"/>
        <filter id="b" width="122.2%" height="138.5%" x="-8.9%" y="-15.4%" filterUnits="objectBoundingBox">
            <feOffset dx="2" dy="2" in="SourceAlpha" result="shadowOffsetOuter1"/>
            <feGaussianBlur in="shadowOffsetOuter1" result="shadowBlurOuter1" stdDeviation="3"/>
            <feColorMatrix in="shadowBlurOuter1" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.114619755 0"/>
        </filter>
    </defs>
```

**Comment by JamieColclough** [comment]
Same Issue here, when using SVGs with filter (top row) and filter with linear gradient (bottom row). This issue leaves me unable to use SVGs for more detailed images.
<img width="98" alt="Screenshot 2020-10-23 at 11 26 53" src="https://user-images.githubusercontent.com/22543944/96993114-c13cf100-1522-11eb-975e-4381e3055e45.png">
